binding.cc

```diff
@@ -390,8 +390,7 @@ namespace zmq {
 
   void
   Socket::UV_PollCallback(uv_poll_t* handle, int status, int events) {
-    assert(status == 0);
-
+    if (status != 0) NanThrowError("I/O status: socket not ready !=0 ");
```

Comment:
these should be `return NanThrowError("I/O status: socket not ready !=0 ");`, else execution will continue